### PR TITLE
Feature/appointment event store

### PR DESCRIPTION
**Pull request by Aleksa1998**

### CODE REVIEW

File: EventStore/EventStore/Service/AppointmentSchedulingEventService.cs [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+        }
+
+        public CountStepsEventDto GetStatisticsMinSteps()
+        {
+            if (GetSuccessfulAppointmentsAttempts().Count > 0)
+            {
+                var minEvent =GetAll().Where(appointmentEvent => GetSuccessfulAppointmentsAttempts().Contains(appointmentEvent.Attempt)).ToList().GroupBy(appointmentEvent => appointmentEvent.Attempt).Select(grp => new { Attempt = grp.Key, Count = grp.Count() - 2 }).ToList().Aggregate((eventWithMinCount, currentEvent) => currentEvent.Count <= eventWithMinCount.Count ? currentEvent : eventWithMinCount);
+                return new CountStepsEventDto(GetEventByAttempt(minEvent.Attempt), minEvent.Count);
+            }
+            return null;
```
Comment (trki293) [inline EventStore/EventStore/Service/AppointmentSchedulingEventService.cs:47]: Write in the form of a ternary operator.

File: EventStore/EventStore/Service/AppointmentSchedulingEventService.cs [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+        }
+
+        public CountStepsEventDto GetStatisticsMaxSteps()
+        {
+            if (GetSuccessfulAppointmentsAttempts().Count > 0)
+            {
+                var maxEvent = GetAll().Where(appointmentEvent => GetSuccessfulAppointmentsAttempts().Contains(appointmentEvent.Attempt)).ToList().GroupBy(appointmentEvent => appointmentEvent.Attempt).Select(grp => new { Attempt = grp.Key, Count = grp.Count() - 2 }).ToList().Aggregate((eventWithMaxCount, currentEvent) => currentEvent.Count >= eventWithMaxCount.Count ? currentEvent : eventWithMaxCount);
+                return new CountStepsEventDto(GetEventByAttempt(maxEvent.Attempt), maxEvent.Count);
+            }
+            return null;
```
Comment (trki293) [inline EventStore/EventStore/Service/AppointmentSchedulingEventService.cs:57]: Write in the form of a ternary operator.
Example:
"var maxEvent = ...;
return ..."

File: EventStore/EventStore/Service/AppointmentSchedulingEventService.cs [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+        public CountStepsEventDto GetStatisticsMinStepsForCancelling()
+        {
+            List<long> unsuccessfulAppointmentsAttempts = GetCanceledAppointmentsAttempts();
+            List<long> successfulAppointmentsAttempts = GetSuccessfulAppointmentsAttempts();
+            if (successfulAppointmentsAttempts.Count > 0)
+            {
+                var minEvent = GetAll().Where(appointmentEvent => unsuccessfulAppointmentsAttempts.Except(successfulAppointmentsAttempts).Contains(appointmentEvent.Attempt)).ToList().GroupBy(appointmentEvent => appointmentEvent.Attempt).Select(grp => new { Attempt = grp.Key, Count = grp.Count() - 1 }).ToList().Aggregate((eventWithMinCount, currentEvent) => currentEvent.Count <= eventWithMinCount.Count ? currentEvent : eventWithMinCount);
+                return new CountStepsEventDto(GetEventByAttempt(minEvent.Attempt), minEvent.Count);
+            }
+            return null;
```
Comment (trki293) [inline EventStore/EventStore/Service/AppointmentSchedulingEventService.cs:89]: Write in the form of a ternary operator.
Example:
"var maxEvent = ...;
return ..."

File: EventStore/EventStore/Service/AppointmentSchedulingEventService.cs [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+        }
+
+        public double GetAverageStepsForUnsuccessfulAttempt()
+        {
+            return (GetCanceledAppointmentsAttempts().Except(GetSuccessfulAppointmentsAttempts()).Count() > 0) ? (((double)GetAll().FindAll(appointmentEvent => GetCanceledAppointmentsAttempts().Except(GetSuccessfulAppointmentsAttempts()).Contains(appointmentEvent.Attempt)).Count() / (double)GetCanceledAppointmentsAttempts().Except(GetSuccessfulAppointmentsAttempts()).Count()) - 1) : 0;
+        }
+
+        public CountStepsEventDto GetStatisticsMinStepsForCancelling()
+        {
+            List<long> unsuccessfulAppointmentsAttempts = GetCanceledAppointmentsAttempts();
```
Comment (trki293) [inline EventStore/EventStore/Service/AppointmentSchedulingEventService.cs:82]: The list instance is called in only one place, call it directly.

File: EventStore/EventStore/Service/AppointmentSchedulingEventService.cs [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+        public CountStepsEventDto GetStatisticsMaxStepsForCancelling()
+        {
+            List<long> unsuccessfulAppointmentsAttempts = GetCanceledAppointmentsAttempts();
+            List<long> successfulAppointmentsAttempts = GetSuccessfulAppointmentsAttempts();
+            if (successfulAppointmentsAttempts.Count > 0)
+            {
+                var maxEvent = GetAll().Where(appointmentEvent => unsuccessfulAppointmentsAttempts.Except(successfulAppointmentsAttempts).Contains(appointmentEvent.Attempt)).ToList().GroupBy(appointmentEvent => appointmentEvent.Attempt).Select(grp => new { Attempt = grp.Key, Count = grp.Count() - 1 }).ToList().Aggregate((eventWithMaxCount, currentEvent) => currentEvent.Count >= eventWithMaxCount.Count ? currentEvent : eventWithMaxCount);
+                return new CountStepsEventDto(GetEventByAttempt(maxEvent.Attempt), maxEvent.Count);
+            }
+            return null;
```
Comment (trki293) [inline EventStore/EventStore/Service/AppointmentSchedulingEventService.cs:101]: Write in the form of a ternary operator.
Example:
"var maxEvent = ...;
return ..."

File: EventStore/EventStore/Service/AppointmentSchedulingEventService.cs [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+            {
+                var minEvent = GetAll().Where(appointmentEvent => unsuccessfulAppointmentsAttempts.Except(successfulAppointmentsAttempts).Contains(appointmentEvent.Attempt)).ToList().GroupBy(appointmentEvent => appointmentEvent.Attempt).Select(grp => new { Attempt = grp.Key, Count = grp.Count() - 1 }).ToList().Aggregate((eventWithMinCount, currentEvent) => currentEvent.Count <= eventWithMinCount.Count ? currentEvent : eventWithMinCount);
+                return new CountStepsEventDto(GetEventByAttempt(minEvent.Attempt), minEvent.Count);
+            }
+            return null;
+        }
+
+        public CountStepsEventDto GetStatisticsMaxStepsForCancelling()
+        {
+            List<long> unsuccessfulAppointmentsAttempts = GetCanceledAppointmentsAttempts();
```
Comment (trki293) [inline EventStore/EventStore/Service/AppointmentSchedulingEventService.cs:94]: The list instance is called in only one place, call it directly.

File: EventStore/EventStore/Service/AppointmentSchedulingEventService.cs [hunk 7]
```diff
@@ -0,0 +1,175 @@
...
+        {
+            return GetSuccessfulAppointmentsAttempts().Count > 0 ? (double)GetAll().Where(appointmentEvent => GetSuccessfulAppointmentsAttempts().Contains(appointmentEvent.Attempt)).ToList().GroupBy(appointmentEvent => appointmentEvent.Attempt).ToList().Select(item => (item.ToList().SingleOrDefault(listItem => listItem.Action.Equals("create")).TimeStamp - item.ToList().SingleOrDefault(listItem => listItem.Action.Equals("start")).TimeStamp).TotalSeconds).Sum() / (double)GetSuccessfulAppointmentsAttempts().Count : 0;
+        }
+        public int GetHowManySchedulesInMinimumSteps()
+        {
+            if (GetSuccessfulAppointmentsAttempts().Count > 0)
+            {
+                return GetAll().Where(appointmentEvent => GetSuccessfulAppointmentsAttempts().Contains(appointmentEvent.Attempt)).ToList().GroupBy(appointmentEvent => appointmentEvent.Attempt).Where(grp => grp.Count() - 2 == 4).ToList().Count();
+            }
+            return 0;
```
Comment (trki293) [inline EventStore/EventStore/Service/AppointmentSchedulingEventService.cs:144]: Write this in the form of a ternary operator.

File: EventStore/EventStore/Service/AppointmentSchedulingEventService.cs [hunk 8]
```diff
@@ -0,0 +1,175 @@
...
+        }
+
+        public int GetHowManySchedulesInMediumSteps()
+        {
+            if (GetSuccessfulAppointmentsAttempts().Count > 0)
+            {
+                return GetAll().Where(appointmentEvent => GetSuccessfulAppointmentsAttempts().Contains(appointmentEvent.Attempt)).ToList().GroupBy(appointmentEvent => appointmentEvent.Attempt).Where(grp => grp.Count() - 2 > 4 && grp.Count() - 2 <= 7).ToList().Count();
+            }
+            return 0;
```
Comment (trki293) [inline EventStore/EventStore/Service/AppointmentSchedulingEventService.cs:153]: Write this in the form of a ternary operator.

File: EventStore/EventStore/Service/AppointmentSchedulingEventService.cs [hunk 9]
```diff
@@ -0,0 +1,175 @@
...
+        }
+
+        public int GetHowManySchedulesInMoreSteps()
+        {
+            if (GetSuccessfulAppointmentsAttempts().Count > 0)
+            {
+                return GetAll().Where(appointmentEvent => GetSuccessfulAppointmentsAttempts().Contains(appointmentEvent.Attempt)).ToList().GroupBy(appointmentEvent => appointmentEvent.Attempt).Where(grp => grp.Count() - 2 > 7).ToList().Count();
+            }
+            return 0;
```
Comment (trki293) [inline EventStore/EventStore/Service/AppointmentSchedulingEventService.cs:162]: Write this in the form of a ternary operator.

**Review by trki293 — CHANGES_REQUESTED** [review]
Necessary changes in order to comply with the principles of Clean code.

**Review by trki293 — APPROVED** [review]
PR is approved. Good Clean code practices are followed.

**Review by velicko-tam — APPROVED** [review]
PR is approved. Clean code principles were implemented.

**Pull request merged**